mm2src/coins/utxo/qtum.rs

```diff
@@ -673,7 +673,7 @@ impl SwapOps for QtumCoin {
         utxo_common::derive_htlc_key_pair(self.as_ref(), swap_unique_data)
     }
 
-    fn derive_htlc_pubkey(&self, swap_unique_data: &[u8]) -> Vec<u8> {
+    fn derive_htlc_pubkey(&self, swap_unique_data: &[u8]) -> [u8; 33] {
```

Comment:
But we already have all these H160/H256/H264/etc.. duplicated types from different crates. For instance we have 2 H160 types one for eth and one for utxo, should we add another kdf internal type? We then have to implement from for all these type, I think it complicates things. I would have gone for associated trait types but this code and others is used in legacy swap code and it would be a huge refactor to change those to associated types.

Reply:
If there are already too many types we should create our own type and wrap them inside it I think. We can also add validation functions and use them instead of manual checks like `if x.len() != 32`. Doing such checks and relying on `[u8; N]` all around the codebase makes it more complicated as it's harder to document and explain in my opinion.